packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -695,8 +710,21 @@ export class Combobox
     if (this.ignoreSelectedEventsFlag) {
       return;
     }
-
     const target = event.target as HTMLCalciteComboboxItemElement["el"];
+
+    const isSelectAllTarget = event
+      .composedPath()
+      .some((node) => (node as HTMLElement).id === `${this.guid}-select-all-enabled`);
```

Comment:
Instead of matching id properties can you store the select all enabled as a reference element to match this?